src/PIL/GifImagePlugin.py

```diff
@@ -915,7 +915,7 @@ def _get_global_header(im, info):
     for extensionKey in ["transparency", "duration", "loop", "comment"]:
         if info and extensionKey in info:
             if (extensionKey == "duration" and info[extensionKey] == 0) or (
-                extensionKey == "comment" and not (1 <= len(info[extensionKey]) <= 255)
+                extensionKey == "comment" and len(info[extensionKey]) == 0
```

Comment:
```suggestion
                extensionKey == "comment" and info[extensionKey]
```
This seems simpler to me?

Reply:
The test needs to determine that `info[comment]` is an empty string to keep `version` as `87a`. Your suggestion `info[extensionKey]` does the opposite. I was attempting to extrapolate from `not (1 <= len(info[extensionKey]) <= 255)` (which originated with PR #1896 but was superseded by PR #3479) to retain the length test. I think it probably could be `info[extensionKey] == ""` or even `not info[extensionKey]`. The 255 was there to begin with, I think, to fit with the maximum length of the single data `block` that was allowed before PR #3479.

Reply:
You're right - I meant `not info[extensionKey]`. That was just a mistake on my part.